src/apply-names.cc

```diff
@@ -32,9 +33,26 @@ namespace wabt {
 
 namespace {
 
-struct Context {
+struct Context : ExprVisitor::DelegateNop {
```

Comment:
My gut tells me we should be using class here since there are many (virtual) methods, but maybe my gut is wrong.